tests/jenkins/TestPrintArtifactDownloadUrlsForStaging.groovy

```diff
@@ -6,23 +6,73 @@
  * compatible open source license.
  */
 
-package jenkins.tests
-
+import jenkins.tests.BuildPipelineTest
 import org.junit.*
 
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.core.IsNull.notNullValue
+
+
+class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest implements LibFunctionTester {
+
+    private List artifactFileNames
+    private String uploadPath
 
-class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest {
 
     @Before
     void setUp() {
+
+        this.artifactFileNames = ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig', 'a_dummy_file.tar.gz']
+        this.uploadPath = 'dummy/upload/path'
+
+        this.registerLibTester(new TestPrintArtifactDownloadUrlsForStaging(
+                artifactFileNames: artifactFileNames,
+                uploadPath: uploadPath
+        ))
+
         super.setUp()
-        binding.setVariable('filenamesForUrls', ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig'])
-        binding.setVariable('UPLOAD_PATH', 'dummy/upload/path')
     }
 
     @Test
     void testPrintArtifactDownloadUrlsForStaging() {
+
+        binding.setVariable('filenamesForUrls', artifactFileNames)
+        binding.setVariable('UPLOAD_PATH', uploadPath)
+
         super.testPipeline("tests/jenkins/jobs/PrintArtifactDownloadUrlsForStaging_Jenkinsfile")
+
+    }
+
+    void configure(helper, bindings){
+    }
+
+    void verifyParams(helper) {
```

Comment:
I think you can use a helper function that would have a calling pattern like this to reduce duplicate code:

```groovy
void verifyParams(helper) {
   verifyCalls(
      methodName: 'printArtifactDownloadUrlsForStaging',
      parameterInvariantsAssertions: call -> {
         assertThat(call.args.artifactFileNames.first(), notNullValue())
         assertThat(call.args.uploadPath.first(), notNullValue())
         assert call.args.artifactFileNames.size() > 0
      }
      expectedParametersMatcher: call -> {
         call.args.uploadPath.first() == this.uploadPath
            && call.args.artifactFileNames.first().sort() == this.artifactFileNames.sort()
      }
   )
}
```